otherlibs/unix/signals.c

```diff
@@ -97,6 +97,19 @@ CAMLprim value caml_unix_sigsuspend(value vset)
   return Val_unit;
 }
 
+CAMLprim value caml_unix_sigwait(value sigs)
+{
+  sigset_t set;
+  int retcode, signo;
+
+  decode_sigset(sigs, &set);
+  caml_enter_blocking_section();
+  retcode = sigwait(&set, &signo);
+  caml_leave_blocking_section();
+  if (retcode == -1) caml_uerror("sigwait", Nothing);
```

Comment:
This logic is wrong. `sigwait` either returns zero or a positive `errno`.
```suggestion
  if (retcode != 0) caml_uerror("sigwait", Nothing);
```

Reply:
Well spotted, thanks.  The correct fix uses `caml_unix_error`, see b62ba96d14